code/osapi/osapi.cpp

```diff
@@ -318,19 +321,31 @@ bool os_is_legacy_mode()
 		checkedLegacyMode = true;
 	}
 	else {
+		bool old_config_exists = false;
+		bool new_config_exists = false;
+
 		SCP_stringstream path_stream;
 		path_stream << getPreferencesPath() << DIR_SEPARATOR_CHAR << Osreg_config_file_name;
 
-		// Use the existance of the fs2_open.ini file for determining if the launcher supports the new mode
-		auto file = fopen(path_stream.str().c_str(), "r");
+		new_config_exists = file_exists(path_stream.str());
+#ifdef SCP_UNIX
+        path_stream.str("");
+		path_stream << Cfile_user_dir_legacy << DIR_SEPARATOR_CHAR << Osreg_config_file_name;
 
-		if (file == nullptr)
-		{
+		old_config_exists = file_exists(path_stream.str());
+#else
+		// At this point we can't determine if the old config exists so just assume that it does
```

Comment:
I will take another look at the code but I was pretty sure that it's not possible without changes to how the config system is initialized. I would like to avoid such changes for this pull request since they would be out of scope.